code/modules/admin/verbs/mapping.dm

```diff
@@ -325,54 +326,78 @@ GLOBAL_VAR_INIT(say_disabled, FALSE)
 	var/list/messages = list()
 	messages += "<b>World</b>: [world.maxx] x [world.maxy] x [world.maxz]<br>"
 
-	var/list/linked_levels = list()
-	var/min_x = INFINITY
-	var/min_y = INFINITY
-	var/max_x = -INFINITY
-	var/max_y = -INFINITY
-
 	for(var/z in 1 to max(world.maxz, z_list.len))
 		if (z > z_list.len)
-			messages += "<b>[z]</b>: Unmanaged (out of bounds)<br>"
+			messages += "Z level: <b>[z]</b>: Unmanaged (out of bounds)<br>"
 			continue
-		var/datum/space_level/S = z_list[z]
-		if (!S)
-			messages += "<b>[z]</b>: Unmanaged (null)<br>"
+		var/datum/space_level/space_level = z_list[z]
+		if (!space_level)
+			messages += "Z level: <b>[z]</b>: Unmanaged (null)<br>"
 			continue
-		var/linkage
-		switch (S.linkage)
-			if (UNAFFECTED)
-				linkage = "no linkage"
-			if (SELFLOOPING)
-				linkage = "self-looping"
-			if (CROSSLINKED)
-				linkage = "linked at ([S.xi], [S.yi])"
-				linked_levels += S
-				min_x = min(min_x, S.xi)
-				min_y = min(min_y, S.yi)
-				max_x = max(max_x, S.xi)
-				max_y = max(max_y, S.yi)
-			else
-				linkage = "unknown linkage '[S.linkage]'"
 
-		messages += "<b>[z]</b>: [S.name], [linkage], traits: [json_encode(S.traits)]<br>"
-		if (S.z_value != z)
-			messages += "-- z_value is [S.z_value], should be [z]<br>"
-		if (S.name == initial(S.name))
-			messages += "-- name not set<br>"
+		messages += "Z level: <b>[z]</b>: [space_level.name]<br>"
+		if (space_level.z_value != z)
+			messages += "-- z_value is [space_level.z_value], should be [z]<br>"
 		if (z > world.maxz)
 			messages += "-- exceeds max z"
 
-	var/grid[max_x - min_x + 1][max_y - min_y + 1]
-	for(var/datum/space_level/S in linked_levels)
-		grid[S.xi - min_x + 1][S.yi - min_y + 1] = S.z_value
-
 	messages += "<table border='1'>"
-	for(var/y in max_y to min_y step -1)
-		var/list/part = list()
-		for(var/x in min_x to max_x)
-			part += "[grid[x - min_x + 1][y - min_y + 1]]"
-		messages += "<tr><td>[part.Join("</td><td>")]</td></tr>"
 	messages += "</table>"
 
 	to_chat(src, messages.Join(""), confidential = TRUE)
+
+#define SUB_ZONE_INFO_FULL(virtual_level) "[virtual_level.parent_map_zone.id]. [virtual_level.id]. [virtual_level.name]"
+#define MAP_ZONE_INFO(map_zone) "[map_zone.id]. [map_zone.name]" //Works for virtual level or map zones
+
+/client/proc/map_zones_info()
+	set name = "Map-Zones Info"
+	set category = "Mapping"
+
+	var/list/dat = list()
+	for(var/datum/map_zone/map_zone as anything in SSmapping.map_zones)
+		dat += "[MAP_ZONE_INFO(map_zone)]:"
+		for(var/datum/virtual_level/virtual_level as anything in map_zone.virtual_levels)
+			dat += "<BR> - [MAP_ZONE_INFO(virtual_level)]:"
+			dat += "<BR> -- Reservation: LowX: [virtual_level.low_x], LowY: [virtual_level.low_y], HighX: [virtual_level.high_x], HighY: [virtual_level.high_y]"
```

Comment:
I know it's a debug tool, but do you think you could include an ADMIN_JMP() link to the low/high turfs? Will probably need a custom href since the macro takes an atom instead of coords

Reply:
ah I missed this comment, yeah I'll do that.